apptail/storage/gob_handler.go

```diff
@@ -0,0 +1,69 @@
+package storage
+
+import (
+	"bytes"
+	"encoding/gob"
+	"github.com/ActiveState/log"
+	"io/ioutil"
+	"os"
+)
+
+// exposing these for testing
+type Storage interface {
+	Encode(data interface{}) ([]byte, error)
+	Load(data interface{})
+	Write(buf []byte) error
+}
+
+type FileStorage struct {
+	file_path string
+}
+
+const FILE_MODE = 0666
+
+func NewFileStorage(path string) Storage {
+	return &FileStorage{
+		file_path: path,
+	}
+
+}
+
+func (s *FileStorage) Encode(data interface{}) ([]byte, error) {
+	m := new(bytes.Buffer)
+	enc := gob.NewEncoder(m)
+	err := enc.Encode(data)
+	if err != nil {
+		return nil, err
+
+	}
+	return m.Bytes(), nil
+}
+
+func (s *FileStorage) Write(buf []byte) error {
+
+	if err := ioutil.WriteFile(s.file_path, buf, FILE_MODE); err != nil {
+		return err
+
+	}
+	// this extra step to make the file accessible by stackato user
+	if err := os.Chmod(s.file_path, FILE_MODE); err != nil {
+		return err
+
+	}
+	return nil
+}
+
+func (s *FileStorage) Load(e interface{}) {
+	n, err := ioutil.ReadFile(s.file_path)
+	if err != nil {
+		log.Error(err)
```

Comment:
Here we log an error because `ReadFile` failed, but continue for process `n` following this conditional. Should we return the err, instead and let the caller decide how they want to proceed?

Would this fail is `s.file_path` did not exist? Seems we assume it always does, but initially it wouldn't.

Would be good to have a test for when the gob file is not present or corrupt.